internal/storage/v1/elasticsearch/spanstore/reader.go

```diff
@@ -616,13 +618,23 @@ func (s *SpanReader) buildFindTraceIDsQuery(traceQuery dbmodel.TraceQueryParamet
 		boolQuery.Must(operationNameQuery)
 	}
 
+	if kind, ok := traceQuery.Tags[spanKindNestedField]; ok {
+		spanKindSpecificQuery := s.buildSpanKindQuery(kind)
```

Comment:
you are missing backwards compatibility here. We should introduce a feature flag that could be on by default but can be turned off by the user if needed. When the feature flag is on we only execute the query against materialized field. When it's off, we do an OR query between top-level field or the nested tag. This is just a rough idea, we need to think through all possible permutations.